### PR TITLE
Fix default rights for self-service profiles

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Internal ref: 19652.

#### Fix default rights for self-service profiles

After creating a new helpdesk profile, the life cycle tabs looks like everything is allowed:
image

After any update, the rights reset to the correct "default" state:
image

To fix this I've added a check in the prepareInputForAdd in order to initialize the life cycle correctly when creating a new helpdesk profile.

No modification needed for central profiles, they already work as expected.

#### Add disabled cursor for readonly checkbox

On the previous screenshot, only 2 out of the 9 checkbox are clickable.
This is very confusing for the users as we give no UI feedback for read-only checkboxes.

After this change any read-only checkbox will display the "not-allowed" cursor when hovered:
image

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: inc/html.class.php [hunk 1]
```diff
@@ -2301,16 +2301,19 @@ static function getCheckbox(array $options) {
          }
       }
 
+      $label_class = "label-checkbox";
+
       if ($params['readonly']) {
          $out .= " disabled='disabled'";
+         $label_class .= " cursor-not-allowed";
```
Comment (orthagh) [inline inc/html.class.php:2308]: instead adding a class, you should rely in your css on the disabled attribute

Comment (AdrienClairembault) [reply inc/html.class.php]: Done in 6c3d48ca6a573b51f88fb79b1910dec8ce551711.

File: inc/profile.class.php [hunk 2]
```diff
@@ -369,6 +369,21 @@ function prepareInputForAdd($input) {
          }
       }
 
+      // Set default values, only needed for helpdesk
+      $interface = isset($input['interface']) ? $input['interface'] : "";
```
Comment (trasher) [inline inc/profile.class.php:373]:
```suggestion
      $interface = $input['interface'] ?? "";
```

File: inc/profile.class.php [hunk 3]
```diff
@@ -369,6 +369,21 @@ function prepareInputForAdd($input) {
...
+      if (!empty($interface) && $interface == "helpdesk" && !isset($input["_cycle_ticket"])) {
+         $tab   = Ticket::getAllStatusArray();
+         $cycle = [];
+         foreach ($tab as $from => $label) {
+            foreach ($tab as $dest => $label) {
```
Comment (trasher) [inline inc/profile.class.php:378]:
```suggestion
            foreach (array_keys($tab) as $dest) {
```

File: inc/profile.class.php [hunk 4]
```diff
@@ -369,6 +369,21 @@ function prepareInputForAdd($input) {
          }
       }
 
+      // Set default values, only needed for helpdesk
+      $interface = isset($input['interface']) ? $input['interface'] : "";
+      if (!empty($interface) && $interface == "helpdesk" && !isset($input["_cycle_ticket"])) {
+         $tab   = Ticket::getAllStatusArray();
+         $cycle = [];
+         foreach ($tab as $from => $label) {
```
Comment (trasher) [inline inc/profile.class.php:377]:
```suggestion
         foreach (array_keys($tab) as $from) {
```

File: inc/profile.class.php (hunk 2 above)
Comment (AdrienClairembault) [reply inc/profile.class.php:373]: This is not possible on PHP 5.6

File: inc/profile.class.php [hunk 5]
```diff
@@ -369,6 +369,21 @@ function prepareInputForAdd($input) {
          }
       }
 
+      // Set default values, only needed for helpdesk
+      $interface = isset($input['interface']) ? $input['interface'] : "";
+      if (!empty($interface) && $interface == "helpdesk" && !isset($input["_cycle_ticket"])) {
```
Comment (cedric-anne) [inline inc/profile.class.php:374]:
```suggestion
      if ($interface == "helpdesk" && !isset($input["_cycle_ticket"])) {
```

**Pull request merged**